data.go

```diff
@@ -456,12 +477,18 @@ func getApidInstanceInfo() (info apidInstanceInfo, err error) {
 	if err != nil {
 		return
 	}
+	tx, err := db.Begin()
+	if err != nil {
+		log.Errorf("getApidInstanceInfo: Unable to get DB tx Err: {%v}", err)
+		return
+	}
 
-	err = db.QueryRow("SELECT instance_id, apid_cluster_id, last_snapshot_info FROM APID LIMIT 1").
+	err = tx.QueryRow("SELECT instance_id, apid_cluster_id, last_snapshot_info FROM APID LIMIT 1").
 		Scan(&info.InstanceID, &savedClusterId, &info.LastSnapshot)
 	if err != nil {
 		if err != sql.ErrNoRows {
 			log.Errorf("Unable to retrieve apidInstanceInfo: %v", err)
+			tx.Rollback()
```

Comment:
Since the error check is complex and nested here, maybe we can try `defer tx.Rollback()` after creating transaction.

Reply:
I am really having mixed feeling using defer for Rollback(). I understand you are handling it, but I simply like to be explicit on purpose, and avoid that code path of execution.

Reply:
FWIW, I prefer the defer as it is simpler. Being explicit is great, but simpler code is more understandable and less likely to have errors or gain errors over time.

Reply:
I like the defer in general as well, as it is simple. But in the txn case, there were issues in the past and I see folks avoid using defer rollback; but should not be an issue anymore.